independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/InjectionPointsTransformer.java

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright 2019 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.quarkus.arc.processor;
+
+import java.lang.annotation.Annotation;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.function.Predicate;
+import org.jboss.jandex.AnnotationInstance;
+import org.jboss.jandex.AnnotationTarget;
+import org.jboss.jandex.AnnotationValue;
+import org.jboss.jandex.DotName;
+import org.jboss.jandex.Type;
+
+/**
+ * Allows a build-time extension to alter qualifiers on an injection point.
+ * <p>
+ *
+ * @author Matej Novotny
+ */
+public interface InjectionPointsTransformer extends BuildExtension {
+
+    /**
+     * Returns true if this transformer is meant to be applied to the supplied {@code typeOfInjectionPoint}.
+     *
+     * @param typeOfInjectionPoint the declared type of the injection point
+     */
+    boolean appliesTo(Type typeOfInjectionPoint);
```

Comment:
I'd prefer to use the same terminology as the CDI spec does: `typeOfInjectionPoint` -> `requiredType`.

Reply:
Fair enough